sootup.core/src/main/java/sootup/core/model/Body.java

```diff
@@ -587,12 +578,12 @@ public String toString() {
    * @param stmts The searched list of statements
    * @return A map of Locals and their using statements
    */
-  public static Map<Local, Collection<Stmt>> collectDefs(Collection<Stmt> stmts) {
-    Map<Local, Collection<Stmt>> allDefs = new HashMap<>();
+  public static Map<LhsValue, Collection<Stmt>> collectDefs(Collection<Stmt> stmts) {
+    Map<LhsValue, Collection<Stmt>> allDefs = new HashMap<>();
     for (Stmt stmt : stmts) {
       List<Value> defs = stmt.getDefs();
       for (Value value : defs) {
-        if (value instanceof Local) {
+        if (value instanceof LhsValue) {
```

Comment:
So, the below code `allDefs.put((Local) value, localDefs);` in line `592`   should be `allDefs.put((LhsValue) value, localDefs);` ?